plugins/modules/aws_ses_identity_dkim.py

```diff
@@ -0,0 +1,169 @@
+            - This should be the email address or domain identifying an existing SES identity
+        required: true
+    state:
+        description:
+            - Enable or disable DKIM on the identity
+        default: enabled
+        choices: [ 'enabled', 'disabled']
+requirements: [ 'botocore', 'boto3' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+
+'''
+
+EXAMPLES = '''
+# Note: to setup SES identities use the aws_ses_identity module.
+
+- name: Enable DKIM setting for identity example@example.com
+  aws_ses_identity_dkim:
+      identity: example@example.com
+
+- name: Enable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+  register: dkim_results
+
+- name: Example how to aws_ses_identity_dkim results to set DNS records for validation (if you use Route53 service for DNS)
+  route53:
+    record: "{{ item }}._domainkey.example.com"
+    value: "{{ item }}.dkim.amazonses.com"
+    type: CNAME
+    zone: example.com
+  with_items: "{{ dkim_results.dkim_attributes.dkim_tokens }}"
+
+- name: Disable DKIM setting for identity example.com
+  aws_ses_identity_dkim:
+    identity: example.com
+    state: disabled
+'''
+
+RETURN = '''
+dkim_attributes:
+    description: Dictionary with DKIM information
+    type: complex
+    returned: always
+    sample: {
+        "dkim_tokens": [],
+        "dkim_enabled": "True",
+        "dkim_verification_status": "Pending"
+    }
+    contains:
+        dkim_tokens:
+            descritpion: Array of the 3 DKIM verification tokens needed for DNS records
+            type: list
+            sample: ['u5d5fa5twxzdni4aaoffdqvgnvku5kvu', '6xzw7n3abimypkyan5s6kuse4nymrfo2', 'nkwigx63eervm4hyja2lwzhjcfqyhka7']
+        dkim_enabled:
+            description: Boolean indicating if DKIM email sending is enabled or disabled for identity
+            type: bool
+            sample: True
+        dkim_verification_status:
+            description: Status of the DKIM verification process (DNS verification)
+            type: str
+            sample: 'Pending'
+'''
+
+from ansible_collections.amazon.aws.plugins.module_utils.aws.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict, AWSRetry
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def get_identity_dkim_settings(module, client, identity):
+    try:
+        response = client.get_identity_dkim_attributes(Identities=[identity])
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to retrieve identity DKIM settings for {identity}'.format(identity=identity))
+    dkim_attributes = response['DkimAttributes']
+
+    return dkim_attributes[identity]
+
+
+def ses_verify_dkim_domain(module, client, identity):
+    domain = identity.split('@')[-1]
+    try:
+        response = client.verify_domain_dkim(Domain=domain)
```

Comment:
using `except is_boto_error_code('SomeErrorIndicatingAbsence'):` it should be possible to catch 'NotFound' and return a null here.  This might make it easier to add check_mode support.

Reply:
ok, will look into that

Reply:
But isn't it better to fail if s.th. goes wrong here? As returning with a null here, will just break anything a user was intending to do with return values.

Reply:
I meant to attach this comment to get_identity_dkim_settings thinking that this was creating a separate object...  If this is just setting a flag on an SES identity, then it's probably better handled as an extension of aws_ses_identity

Reply:
That's a valid point. I guess I just started a seperate module back then because I thought there is not much overlap between the two modules in terms of logic and params needed. But as both modules use the same Boto3 module one could also argue to keep things together. Not sure what the general policy is for AWS modules. As it seems newer modules rather go the one purpose one module approach, but I could be wrong.

We should probably decide on this first before putting more effort into this PR.